### PR TITLE
Integrate Graphql API for stats page

### LINKED ISSUE #532: [SPEC] DApp GraphQl integration for the stats page
## Overview

Integration spec for the GraphQL API in the dApp, for the status usage the dApp wil just pull the data without any
mutation to them.

**The goals are:**

1. Integrating the GraphQl API and the codegen steps
2. Ensuring the caching strategy for the data as there other areas on our _origin_ that consumes localstorage
3. Implement the stats page parts the deal with the graphql API
4. Making the graphql/stats standalone from the dApp provider

## Checklist

- Provider
    - [ ] Start the `WebbProvider` lazily if the `stats` App is integrated on the dApp as page &load only the required
      dApp code to run the stats page
    - [ ] There's could be a config for graphql backend to consume with some metadata (as the stats is captured from multiple SubQuery nodes)
  ```typescript
  export type GqlBackendConfig = {
    uri:string;
    chainId:ParsedChainId;
  } 
  - ```
- Codegen
    - [ ] Integrate the `codgen.yml` for [Graphql Code Generator](https://www.graphql-code-generator.com/)
```yml

overwrite: true
schema: 'PATH_OF_SUB_QUERY_NODES'
documents: 'apps/stats/**/*.graphql'
generates:
  apps/stats/components/graphql.tsx:
    plugins:
      - 'typescript'
      - 'typescript-operations'
      - 'typescript-react-apollo'
  ./graphql.schema.json:
    plugins:
      - 'introspection'
    config:
      withComponent: true
      withHooks: true
      withMutationFn: true
      withRefetchFn: true
      skipTypename: true

```
### Structure

As the statistics are just about data fetching only the Graphql query is needed.

1. Creating a new dApp package `stats` which will be a standalone package, and won't consume any of other dApp's
   providers. it will expose the graphql query components, the types of the GraphqlEntities, and a dApp page that can be
   integrated in the dApp, or served as a standalone page.
2. Making the dApp provider instantiation lazy so if the `stats` page is active we don't necessarily instantiate the
   provider.
3. Graphql query for an entity

```graphql
# /apps/stats/src/graphql/queries.gql
query NextAuthorities {
    authorites {
        nodes {
            id
            next
            block {
                id
                number
            }
        }
    }
}
```

2. Codegen Generation will generate a ReactComponent for the query

```typescript jsx
import { Authority, NextAuthorities } from "@webb-dapp/stats";

const NextAuthrotiesStats = () => {
  const [selectedAuthority, setSelectedAuthority] = useState<Authority | null>(null);
  return (
    <NextAuthorities>
      {({ data, loading, error }) => {
        if (loading) return <p>Loading next authorites</p>;
        if (error) return <p>Error :(</p>;
        return <button onClick={() => {
          setSelectedAuthority(data.authorites.nodes[0]);
        }}>{data.authorites.nodes[0].next}</button>
      }}
    </NextAuthorities>
  );
};
```

## Resources

- [Graphql Code Generator](https://www.graphql-code-generator.com/)
- [React Apollo](https://www.apollographql.com)

### LINKED ISSUE #550: [TASK] Add script to update all `webb.js` dependencies to a version
# Overview

Add a node script that updates the root package.json resolutions and all package.json dependencies

### LINKED ISSUE #554: [TASK] Updated the dApp with  `protocol-solidity` new types
## Overview
Update dApp with latest types changes for `ExtData` ,proof inputs
**Places to update**
1. `VAnchorContract`
2. `PolkadotVAnchorDeposit`
3. `PolkadotVAnchorWithdraw`
4. `Web3VAnchorWithdraw`

### PR DESCRIPTION
**Pull request by AhmedKorim**
## Summary of changes
- Integrate the GraphQl API
- Update Polkadot packages
- Update to the latest webb.js
- Update to latest protocol-solity
- Added script to update packages versions 

image

### Reference issue to close (if applicable)
- Closes #532 
- Closes #554 
- Closes #550 

-----
### Code Checklist 

- [X] Tested
- [x] Documented

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,6 +79,14 @@
...
   "devDependencies": {
     "@babel/plugin-syntax-import-meta": "^7.10.4",
+    "@graphql-codegen/cli": "2.12.0",
+    "@graphql-codegen/fragment-matcher": "3.3.1",
+    "@graphql-codegen/introspection": "2.2.1",
+    "@graphql-codegen/typescript": "2.7.3",
+    "@graphql-codegen/typescript-document-nodes": "2.3.3",
+    "@graphql-codegen/typescript-graphql-files-modules": "2.2.1",
+    "@graphql-codegen/typescript-operations": "2.5.3",
+    "@graphql-codegen/typescript-react-apollo": "3.3.3",
```
Comment (AtelyPham) [inline package.json:89]: Some plugins you have commented out in the plugins section of the `codegen.yml` file, can we remove any unused dependencies here?

File: packages/api-providers/src/contracts/wrappers/webb-vanchor.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import type { JsNote } from '@webb-tools/wasm-utils';
 
 import { Log } from '@ethersproject/abstract-provider';
+import { BytesLike } from '@ethersproject/bytes';
```
Comment (nepoche) [inline packages/api-providers/src/contracts/wrappers/webb-vanchor.ts:8]:
```suggestion
```

File: packages/api-providers/src/polkadot/vanchor-deposit.ts [hunk 3]
```diff
@@ -190,6 +190,8 @@ export class PolkadotVAnchorDeposit extends VAnchorDeposit<WebbPolkadot, Deposit
         extAmount: BigNumber.from(publicAmount),
         encryptedOutput1: u8aToHex(comEnc1),
         encryptedOutput2: u8aToHex(comEnc2),
+        refund: '0',
+        token: '1',
```
Comment (nepoche) [inline packages/api-providers/src/polkadot/vanchor-deposit.ts:194]: The `extData` value should be consistent with the value in the `ProvingManagerSetupInput`. I would suggest a change to both for something like:

extData - wrappableAssetRaw.toString() ?? '0'
provingInput - hexToU8a(wrappableAssetRaw.toString()) ?? Uint8Array.from([0])

File: packages/api-providers/src/polkadot/vanchor-withdraw.ts [hunk 4]
```diff
@@ -225,6 +227,8 @@ export class PolkadotVAnchorWithdraw extends VAnchorWithdraw<WebbPolkadot> {
             fee: extData.fee,
             encryptedOutput1: Array.from(hexToU8a(extData.encryptedOutput1)),
             encryptedOutput2: Array.from(hexToU8a(extData.encryptedOutput2)),
+            refund: '0',
+            token: u8aToHex(Uint8Array.from([0])),
```
Comment (nepoche) [inline packages/api-providers/src/polkadot/vanchor-withdraw.ts:231]:
```suggestion
            token: '0',
```

File: packages/page-governance/src/ProposalDetail/VotersTable/VotersTable.tsx [hunk 5]
```diff
@@ -2,12 +2,13 @@ import { Typography } from '@mui/material';
...
 import { useColorPallet } from '@webb-dapp/react-hooks/useColorPallet';
 import { Config, UserConfig } from 'gridjs';
-import { Grid } from 'gridjs-react';
+import { Grid as _Grid } from 'gridjs-react';
 import React, { useMemo } from 'react';
 
 import { Wrapper } from '../CastVote/styled';
 import { IProposalVoter } from '../useProposalDetail';
 import { ChipWrapper } from './styled';
+const Grid = _Grid as any;
```
Comment (nepoche) [inline packages/page-governance/src/ProposalDetail/VotersTable/VotersTable.tsx]: Please add a comment as to why this is how we are importing from 'gridjs-react'

**Review by nepoche — CHANGES_REQUESTED** [review]

**Review by nepoche — APPROVED** [review]

**Pull request merged**